docs/howto/manage-web-app-charms/use-a-database.rst

```diff
@@ -0,0 +1,335 @@
+  `Alembic <https://alembic.sqlalchemy.org/en/latest/>`_.
+
+
+Prepare the environment
+-----------------------
+
+MySQL requires a specific Python driver. Ensure your ``requirements.txt`` includes
+a MySQL driver compatible with SQLAlchemy, such as ``pymysql``.
+
+.. code-block::
+    :caption: requirements.txt
+
+    fastapi
+    sqlalchemy
+    alembic
+    PyMySQL[rsa]
+
+.. warning::
+
+  When using ``pymysql``, your connection string (handled later by the charm)
+  generally follows the format ``mysql+pymysql://user:pass@host/db``.
+
+
+Declare models
+--------------
+
+First, define the structure of your data. We will create a simple ``User``
+model using SQLAlchemy's declarative system.
+
+Create a file named ``models.py``:
+
+
+.. code-block:: python
+    :caption: requirements.txt
+
+    from sqlalchemy import Column, Integer, String
+    from sqlalchemy.orm import declarative_base
+
+    Base = declarative_base()
+
+    class User(Base):
+        __tablename__ = "users"
+
+        id = Column(Integer, primary_key=True, index=True, autoincrement=True)
+        # MySQL VARCHAR requires a length
+        username = Column(String(50), unique=True, index=True, nullable=False)
+        email = Column(String(100), unique=True, index=True, nullable=False)
+
+
+Define migrations
+-----------------
+
+Database schemas change over time. To manage these changes without losing data,
+let's use **Alembic** for migrations.
+
+Initialize Alembic
+~~~~~~~~~~~~~~~~~~
+
+In your project root, run:
+
+.. code-block:: bash
+
+    alembic init alembic
+
+Configure the environment
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Edit ``alembic/env.py`` to point to your models so Alembic can detect changes.
+You also need to configure it to read the database URL from the environment,
+adhering to 12-factor principles.
+
+
+.. code-block:: python
+    :caption: alembic/env.py
+
+    import os
+    from logging.config import fileConfig
+    from sqlalchemy import engine_from_config, pool
+    from alembic import context
+    # Import your models specifically
+    from models import Base
+
+    config = context.config
+
+    # 12-factor app: Read config from environment variable
+    # The Juju integration will provide this variable
+    db_url = os.getenv("MYSQL_DB_CONNECT_STRING")
+
+    if db_url:
+        # Convert mysql:// to mysql+pymysql:// for compatibility
+        if db_url.startswith("mysql://"):
+            db_url = db_url.replace("mysql://", "mysql+pymysql://", 1)
+        config.set_main_option("sqlalchemy.url", db_url)
+
+    target_metadata = Base.metadata
+
+    # ... rest of the file ...
+
+Generate a migration
+~~~~~~~~~~~~~~~~~~~~
+
+Create the revision file that instructs the database how to create the ``users`` table.
+This command requires us to connect to the database, but since we do not have access
+to the database, we will use SQLite instead by setting the enviroment variable to
+a non-existent SQLite instance:
```

Comment:
> we will use SQLite

Is "use" appropriate? To less attentive readers this might sound like we're bringing in another DB system.